### PR TITLE
add address to customer

### PR DESCRIPTION
**Pull request by darmhoo**
get address through geolocation and store to db

### CODE REVIEW

File: app/api/customer/route.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { NextResponse } from "next/server";
```
Comment (teebarg) [inline app/api/customer/route.js:1]: Don't we have this route already?

File: app/user/[id]/verify-location/page.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+      const response = await fetch('/api/save-address', {
+          method: 'POST',
+          body: JSON.stringify({ id: params.id, add_street: loc, add_addinfo_description: loc })
+      })
+
+      await response.json()
+      router.push(`/user/${params.id}`);
+  } catch (error) {
+    console.log(error)
```
Comment (teebarg) [inline app/user/[id]/verify-location/page.js]: remove log

File: app/user/[id]/verify-location/page.js [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+      })
+
+      return resp;
+  }
+
+  const getLocation = () => {
+    if (navigator.geolocation) {
+      navigator.geolocation.getCurrentPosition(showPosition)
+    } else {
+      console.log("Geolocation is not supported by this browser.")
```
Comment (teebarg) [inline app/user/[id]/verify-location/page.js]: remove console

File: app/user/[id]/verify-location/page.js [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+          "User-Agent": "Thunder Client (https://www.thunderclient.com)",
+        },
+      }
+    )
+      .then((response) => {
+        return response.json()
+      })
+      .then((res) => {
+       let  data = res.results[0].formatted
+        console.log(data)
```
Comment (teebarg) [inline app/user/[id]/verify-location/page.js]: remove console

File: app/api/customer/route.js (hunk 1 above)
Comment (darmhoo) [reply app/api/customer/route.js:1]: That one is to get all customers, i only need a specific one

File: app/user/[id]/verify-location/page.js [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+  async function showPosition(position) {
+    let res = await getInWords(
+      position.coords.longitude,
+      position.coords.latitude
+    )
+    setLoc(res);
+  }
+
+  async function getInWords(longitude, latitude) {
```
Comment (cheedinmar) [inline app/user/[id]/verify-location/page.js]: what does this function do?

File: app/api/customer/route.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+import { queryPromise, pool } from "@/utils/sql";
+
+
+export async function GET(request) {
```
Comment (cheedinmar) [inline app/api/customer/route.js:5]: What is this function getting? Can we use GET_CUSTOMER

File: app/api/save-address/route.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
+import { NextResponse } from "next/server";
+import { queryPromise, pool } from "@/utils/sql";
+
+
+export async function POST(request) {
```
Comment (cheedinmar) [inline app/api/save-address/route.js:5]: is it possible to change these functions name to SAVE_ADDRESS

File: app/api/customer/route.js (hunk 6 above)
Comment (darmhoo) [reply app/api/customer/route.js:5]: It is like the naming convention of nextjs server side, showing that it is a get request for the customer route. U can verify with the nextjs doc. So a post request here will have POST as function

File: app/api/save-address/route.js (hunk 7 above)
Comment (darmhoo) [reply app/api/save-address/route.js:5]: The same thing applies here

File: app/user/[id]/verify-location/page.js (hunk 5 above)
Comment (darmhoo) [reply app/user/[id]/verify-location/page.js]: it converts the co-ordinate to human readable physical address

**Review by teebarg — APPROVED** [review]

**Pull request merged**